File: roteiro_03/java_logic/desafios/desafio5.java
package roteiro_03.java_logic.desafios;

import java.util.Scanner;
public class desafio5 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        String senhaCorreta="1234";
        String senhaDigitada="";

        while(!senhaDigitada.equals(senhaCorreta)){
            System.out.println("Digite a senha para entrar:");
            senhaDigitada=sc.nextLine();

            if (!senhaDigitada.equals(senhaCorreta)){
                System.out.println("Senha incorreta! Tente novamente");
            }
        }
        System.out.println("Login realizado com sucesso!!! Bem-vindo");
        sc.close();
    }
    
}

File: roteiro_03/cpp_logic/desafio5.cpp
#include <iostream>
#include <string> 
using namespace std;

int main() {
    string senha_correta = "senha123";
    string tentativa = "";

    cout << "--- Tela de Login ---" << endl;

    while (tentativa != senha_correta) {
        cout << "Digite a senha: ";
        cin >> tentativa;

        if (tentativa != senha_correta) {
            cout << "Senha incorreta. Tente novamente.\n" << endl;
        }
    }

    cout << "\nAcesso permitido! Boas-vindas ao sistema." << endl;

    return 0;
}